src/components/Dropdown/Dropdown.tsx

```diff
@@ -786,9 +810,47 @@ const Dropdown = ({
     [rememberScrollPosition],
   );
 
-  const closeIcons = (
-    <>
-      {handleClear && values.length > 0 && (
+  const valueCountCloseIconHandler = () => {
+    if (showValueCount && showCloseIcon) {
+      return (
+        <>
+          <StyledValueCountContainer
+            variant={valueCountVariant}
+            color={defaultedColor}
+            ref={valueCountRef}
+            dropdownVariant={variant}
+            {...valueCountProps}
+          >
+            {values.length}
+          </StyledValueCountContainer>
+          <StyledCloseIconContainer
+            onMouseDown={(e: React.FocusEvent) => e.stopPropagation()}
+            onClick={handleClear}
+            onFocus={(e: React.FocusEvent) => e.stopPropagation()}
+            tabIndex={tabIndex}
+            ref={closeIconRef}
+            {...closeIconProps}
+          >
+            <Icon path={mdiClose} size="1em" />
+          </StyledCloseIconContainer>
+        </>
+      );
+    }
+    if (showValueCount && !showCloseIcon) {
```

Comment:
I think you're way overcomplicating this lol

Reply:
Yeah the if block seemed like there had to be a better way to me.. Looking into it.